pyspark_ai/pyspark_ai.py

```diff
@@ -253,8 +253,10 @@ def _create_dataframe_with_llm(
         self._spark.sql(sql_query)
         return self._spark.table(view_name)
 
-    def _get_df_schema(self, df: DataFrame) -> str:
-        return "\n".join([f"{name}: {dtype}" for name, dtype in df.dtypes])
+    def _get_df_schema(self, df: DataFrame) -> Tuple[list, str]:
```

Comment:
let's split this into two methods. Returning two values here are confusing

Reply:
We can have `_get_df_schema` which returns a list;
to get schema_str, we can just join the result from `_get_df_schema`